src/com/pentalog/pentastagiu/homework2/shop/Basket.java

```diff
@@ -0,0 +1,53 @@
+package com.pentalog.pentastagiu.homework2.shop;
+
+
+import com.pentalog.pentastagiu.homework2.shop.customermembership.MembershipType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Basket {
+    private int totalPrice;
+    private List<Product> products;
+
+    public Basket() {
+        products = new ArrayList<>();
+    }
+
+    public void addProduct(Product product) {
+        totalPrice += product.getPrice();
+        products.add(product);
+    }
+
+    public int getTotalFinalPrice(MembershipType membershipType) {
+        int finalPrice = 0;
+        switch (membershipType) {
+            case GOLD:
```

Comment:
pt a fi usor de extins pe viitor, procentul asociat unui tip de membership ar trebui pastrat direct in enum(un private field si un getter public pe care il vei folosi in orice clasa in care vei vrea sa te folosesti de discount pentru un anumit tip de membership); astfel daca ai folosi in mai multe locuri din proiect conceptul de discount si el ar fi implementat in enum, intr-un singur loc, vei putea sa faci usor orice modificare fara sa fie nevoie sa mai schimbi alte bucati de cod